lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -91,6 +114,84 @@ impl ShardReplicaSet {
             return Ok(false);
         }
 
+        let segments_path = LocalShard::segments_path(replica_path);
+
+        let mut snapshot_segments = HashSet::new();
+        let mut snapshot_manifests = SegmentManifests::default();
+
+        for segment_entry in segments_path.read_dir()? {
+            let segment_path = segment_entry?.path();
+
+            if !segment_path.is_dir() {
+                log::warn!(
+                    "segment path {} in extracted snapshot {} is not a directory",
+                    segment_path.display(),
+                    replica_path.display(),
+                );
+
+                continue;
+            }
+
+            let segment_id = segment_path
+                .file_name()
+                .and_then(|segment_id| segment_id.to_str())
+                .expect("segment path ends with a valid segment id");
+
+            let added = snapshot_segments.insert(segment_id.to_string());
+            debug_assert!(added);
+
+            let manifest_path = segment_path.join("segment_manifest.json");
+
+            if recovery_type.is_full() {
+                if manifest_path.exists() {
```

Comment:
Should be the same, but maybe this makes it a little bit more explicit

```suggestion
                if manifest_path.is_file() {
```

Reply:
> even if segment_manifest.json is not a file

When can this happen? :thinking:

Reply:
> Never, but just in case? Debugging (even trivial) snapshots and file related bugs is extremely annoying, so I just check/log/contextualize everything.